queries.sql

```diff
@@ -1,11 +1,60 @@
-
 /*Queries that provide answers to the questions from all projects.*/
 
 SELECT * FROM animals WHERE name LIKE '%mon';
+
 SELECT name FROM animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
-SELECT name FROM animals WHERE neutered = true AND escape_attempts < 3;
+
+SELECT name FROM animals WHERE neutered = True AND escape_attempts < 3;
+
 SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
-SELECT * FROM animals WHERE neutered = true; 
-SELECT * FROM animals WHERE name NOT IN ('Gabumon');
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+SELECT * FROM animals WHERE neutered = True;
+
+SELECT * FROM animals WHERE name != 'Gabumon';
+
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+--Begin animal transaction
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+SELECT * FROM animals;
+
+-- another animal transaction
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+
+
+-- delete transaction
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+-- another delete transaction
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SAVEPOINT_1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT_1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+
+-- queries to answer following questions
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```

Comment:
- [ ] Kindly make sure you correctly answer to this query as you are supposed to get the `highest` `escape_attempts` and not the `total` of the `escape_attempts`. If you correctly read the query, it is said: `Who escapes the most, neutered or not neutered animals?`, that means you can use `MAX` instead of `SUM` for example:
```sql
..., MAX(escape_attempts) FROM animals ...
```